cobbler/items/item.py

```diff
@@ -22,9 +22,71 @@
 from cobbler.cexceptions import CX
 from cobbler.decorator import InheritableProperty, InheritableDictProperty
 
+
 RE_OBJECT_NAME = re.compile(r"[a-zA-Z0-9_\-.:]*$")
 
 
+class ItemCache:
+    """
+    A Cobbler ItemCache object.
+    """
+
+    def __init__(self, api):
+        """
+        Constructor
+        """
+        # Parameterized cache for to_dict(resolved: bool).
+        # The values of the resolved parameter are the key for the Dict.
+        self._cached_dict: Dict[bool, Optional[Dict]] = {True: None, False: None}
```

Comment:
Since we don't need a magic property that is compact and have a dedicated object for this purpose, I would like to see this split up. This suggestion is opinionated but I like explicit code.

```suggestion
        self._cached_dict_raw: Optional[Dict] = None
        self._cached_dict_resolved: Optional[Dict] = None
```

Reply:
In theory, a parametrized cache is a Dict with the following structure:
```
   cache_key        cache_value
{ (P1, P2, .., Pn): value }
```
where P1, .., Pn are cache parameters

Therefore, operations with the cache in the general case look like this:
```python
    cache: Dict[Any, Any] = {}

def get_cache(p1, .., pn):
    if (p1, .., pn) in cache:
        return cache[(p1, .., pn)]

def set_cache(p1, .., pn, value):
        cache[(p1, .., pn)] = value

def clear_cache(p1, .., pn, value):
    if (p1, .., pn) in cache:
        del cache[(p1, .., pn)]
```
The magic in the `to_dict` case is that there is only one parameter (`resolved`) and only 2 parameter values (`bool`). Because of this, I can avoid creating and deleting keys for the cache each time and get rid of two "if" by creating all key values at initialization.

I could agree to bring the code to a general form at the expense of less efficiency., but the idea of separate variables `cached_dict_raw` and `cached_dict_resolved` seems wrong to me.

Reply:
As much as this is un-intuitive imho, this PR was created because we are not efficient enough. Thus my concerns should be seen as overruled. Thanks for the explanation.

Reply:
Added more comments https://github.com/cobbler/cobbler/pull/3266/commits/d8117a8cb4f51494abf7ebc60d0d9421704cf042